### ISSUE TITLE
Incorrect query results when sorting on a relationship with no edges

### ISSUE THREAD

**Issue by dennisjlee** [neo4j/graphql#2262]
**Describe the bug**
I found a regression in @neo4j/graphql 3.9.0 and higher, where adding a sort in a graphql query results in losing data from the result set. This happens when sorting a `connection` field where there happened to be no matching relationships.

**Type definitions**
```
    type Component {
        uuid: String
        upstreamProcess: Process @relationship(type: "OUTPUT", direction: IN)
        downstreamProcesses: [Process!]! @relationship(type: "INPUT", direction: OUT)
    }

    type Process {
        uuid: String
        componentOutputs: [Component!]! @relationship(type: "OUTPUT", direction: OUT)
        componentInputs: [Component!]! @relationship(type: "INPUT", direction: IN)
    }
```

**To Reproduce**
Steps to reproduce the behavior: see README in [this repo](https://github.com/dennisjlee/neo4j-graphql-bug-repro/tree/cd093d949c4579d8b0f4cba4de6588faf307e919) with a minimal reproduction

TL;DR: this query behaves incorrectly

```
query ComponentsProcesses {
  components(where: {uuid: "c1"}) {
    uuid
    upstreamProcessConnection {
      edges {
        node {
          uuid
          componentInputsConnection(sort: [{node: { uuid: DESC }}]) {
            edges {
              node {
                uuid
              }
            }
          }
        }
      }  
    }
  }
}
```

**Expected behavior**
In the case that I found, there is one relationship for `upstreamProcessConnection`, and then zero relationships from there in `componentInputsConnection`. So I would expect to get a result set back that has one edge and node under `upstreamProcessConnection` and zero edges/nodes under `componentInputsConnection`. 

Instead, the buggy behavior is that `upstreamProcessConnection` returns zero edges/nodes.

**System (please complete the following information):**

* OS: Windows
* Version: @neo4j/graphql@3.10.1 (also reproduced on 3.9.0, but working on 3.8.0)
* Node.js version: v16.14.0

**Additional context**

Here is the generated Cypher for this snippet of graphql.
```
query ComponentsProcesses2 {
  buggyComponents: components(where: {uuid: "c1"}) {
    uuid
    upstreamProcessConnection {
      edges {
        node {
          uuid
          componentInputsConnection(sort: [{node: {uuid: DESC}}]) {
            edges {
              node {
                uuid
              }
            }
          }
        }
      }
    }
  }
}
```

Version 3.10.1 (buggy)
```
MATCH (this:`Component`)
WHERE this.uuid = $param0
CALL {
    WITH this
    MATCH (this)<-[this_connection_upstreamProcessConnectionthis0:OUTPUT]-(this_Process:`Process`)
    CALL {
        WITH this_Process
        MATCH (this_Process)<-[this_Process_connection_componentInputsConnectionthis0:INPUT]-(this_Process_Component:`Component`)
        WITH this_Process_connection_componentInputsConnectionthis0, this_Process_Component
        ORDER BY this_Process_Component.uuid DESC
        WITH { node: { uuid: this_Process_Component.uuid } } AS edge
        WITH collect(edge) AS edges
        WITH edges, size(edges) AS totalCount
        UNWIND edges AS edge
        WITH edge, totalCount
        ORDER BY edge.node.uuid DESC
        WITH collect(edge) AS edges, totalCount
        RETURN { edges: edges, totalCount: totalCount } AS this_Process_componentInputsConnection
    }
    WITH { node: { uuid: this_Process.uuid, componentInputsConnection: this_Process_componentInputsConnection } } AS edge
    WITH collect(edge) AS edges
    WITH edges, size(edges) AS totalCount
    RETURN { edges: edges, totalCount: totalCount } AS this_upstreamProcessConnection
}
RETURN this { .uuid, upstreamProcessConnection: this_upstreamProcessConnection } as this
Params:
{
  "param0": "c1"
}
```

Version 3.8.0 (working)
```
MATCH (this:`Component`)
WHERE this.uuid = $param0
CALL {
    WITH this
    MATCH (this)<-[this_output_relationship:OUTPUT]-(this_process:Process)
    CALL {
        WITH this_process
        MATCH (this_process)<-[this_process_input_relationship:INPUT]-(this_process_component:Component)
        WITH this_process_input_relationship, this_process_component
        ORDER BY this_process_component.uuid DESC
        WITH collect({ node: { uuid: this_process_component.uuid } }) AS edges
        UNWIND edges as edge
        WITH edges, edge
        ORDER BY edge.node.uuid DESC
        WITH collect(edge) AS edges, size(collect(edge)) AS totalCount
        RETURN { edges: edges, totalCount: totalCount } AS componentInputsConnection
    }
    WITH collect({ node: { uuid: this_process.uuid, componentInputsConnection: componentInputsConnection } }) AS edges
    UNWIND edges as edge
    WITH collect(edge) AS edges, size(collect(edge)) AS totalCount
    RETURN { edges: edges, totalCount: totalCount } AS upstreamProcessConnection
}
RETURN this { .uuid, upstreamProcessConnection } as this
Params:
{
  "param0": "c1"
}
```

**Comment by neo4j-team-graphql** [comment]
Many thanks for raising this bug report @dennisjlee. :bug: We will now attempt to reproduce the bug based on the steps you have provided.

Please ensure that you've provided the necessary information for a minimal reproduction, including but not limited to:
* Type definitions
* Resolvers
* Query and/or Mutation (or multiple) needed to reproduce

If you have a support agreement with Neo4j, please link this GitHub issue to a new or existing Zendesk ticket.

Thanks again! :pray:

**Comment by dennisjlee** [comment]
A couple more notes - here are some variations on the initial scenario that _do_ work as expected in v3.10.1:
* if there are relationships in the database for `componentInputsConnection`
* if we remove the sorting on `componentInputsConnection`
* if we query using `componentInputs` instead of `componentInputsConnection`, like this:
```
components(where: { uuid: "c1" }) {
    uuid
    upstreamProcessConnection {
      edges {
        node {
          uuid
          componentInputs(options: {sort: [{ uuid: DESC }]}) {
            uuid
          }
        }
      }
    }
  }
```

**Comment by neo4j-team-graphql** [comment]
We've been able to confirm this bug using the steps to reproduce that you provided - many thanks @dennisjlee! :pray: We will now prioritise the bug and address it appropriately.

**Comment by angrykoala** [comment]
Hi, @dennisjlee Thanks for the thorough report. This is fixed in the PR #2268 and will be published in a new release soon. Sorry for the inconvenience caused by the regression

**Comment by dennisjlee** [comment]
Hi @angrykoala  - thank you for the quick fix!
